kernel/kernel-api/src/main/java/io/delta/kernel/internal/table/SnapshotFactory.java

```diff
@@ -60,14 +63,23 @@ public static long resolveTimestampToSnapshotVersion(
       Engine engine,
       SnapshotQueryContext snapshotQueryCtx,
       SnapshotImpl latestSnapshot,
-      long millisSinceEpochUTC) {
+      long millisSinceEpochUTC,
+      List<ParsedLogData> logDatas) {
+
+    // Filter out only the ratified staged commits as that is all that DeltaHistoryManager supports
+    List<ParsedDeltaData> parsedDeltaDatas =
+        logDatas.stream()
+            .filter(logData -> logData instanceof ParsedDeltaData)
+            .map(logData -> (ParsedDeltaData) logData)
+            .filter(deltaData -> deltaData.isFile() && deltaData.isRatifiedCommit())
```

Comment:
Eventually we will support inline commit right?

Reply:
maybe consider throw at some layer in case, uc side suddenly supports inline commit and kernel just filter it out and return partial data.

Reply:
I will defer to @scottsand-db on this since I don't think this is unique to here (but rather all places we accept logDatas?)

Reply:
Sorry -- more detailed answer, this is impossible currently, we validate in SnapshotBuilderImpl that we only accept `isFile` https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/table/SnapshotBuilderImpl.java#L182

Reply:
I updated this code to instead validate that we still only have staged ratified commits as is validated in SnapshotBuilderImpl. This way, if SnapshotBuilderImpl ever changes, this code should fail.